### PR TITLE
[xcode11] [ItunesLibrary] macOS b1 bindings

### PR DESCRIPTION
**Pull request by joconte**
diff : https://github.com/xamarin/xamarin-macios/wiki/iTunesLibrary-macOS-xcode11-beta1

It probably needs more work but we can discuss about it here 😄

### CODE REVIEW

**Comment by joconte** [comment]
@mandel-macaque Oh It's today ! Yeah I will double check.

**Comment by joconte** [comment]
Well, according to the wiki, nothing changed in beta 4.

**Comment by dalexsoto** [comment]
build

**Comment by joconte** [comment]
Ouch build failed

**Comment by spouliot** [comment]
@joconte there's an [example](https://github.com/xamarin/xamarin-macios/blob/master/src/CoreLocation/CLCompat.cs) and the new file needs to be added to `src/frameworks.sources`

**Comment by spouliot** [comment]
The build error is a known random issue -> https://github.com/xamarin/maccore/issues/1888

**Comment by joconte** [comment]
@spouliot Wow thanks for the detailed answer. I will fix it.

**Comment by joconte** [comment]
@spouliot What about this ?

File: src/iTunesLibrary/ITCompat.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace iTunesLibrary
+{
+#if !XAMCORE_2_0
```
Comment (spouliot) [inline src/iTunesLibrary/ITCompat.cs]: close :) I should have mentioned it but `XAMCORE_2_0` is a few years back (when we did 32/64 unified)

What you need if `#if XAMCORE_4_0` so, the next time we read API compatibility (if ever), that code can be removed.

and, just to paint a complete picture, `XAMCORE_3_0` was used to exclude stuff when we added tvOS and watchOS support (so we did not bring `[Obsolete]`d API in the new platforms.

File: src/iTunesLibrary/ITCompat.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace iTunesLibrary
+{
+#if !XAMCORE_2_0
+    public partial class ITLibrary {
+
+        [Obsolete ("Does not return a valid instance on macOS 14.")]
```
Comment (spouliot) [inline src/iTunesLibrary/ITCompat.cs]: There's no need to mention the version - it's not clear if this a change in macOS 10.15 (or if it was already the case) and the type is new in macOS 10.14.

Also it's not `macOS 14` ;-) even if I wish that least year `X` removal (from Mac OSX) also removed the `10`

-> `[Obsolete ("This constructor does not create a valid instance of the type.")]`

^ it's an existing string (so it won't increase the metadata size)

File: src/iTunesLibrary/ITCompat.cs (hunk 1 above)
Comment (joconte) [reply src/iTunesLibrary/ITCompat.cs]: Thanks a lot for the detailed explanations, I'm learning so much about the project everyday !

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
Fixed :) If the bots are happy (and they should be) then it can be merged. Thanks!

**Comment by joconte** [comment]
Ahhh I forgot to delete the todo file

**Comment by joconte** [comment]
I will to do it later today

**Comment by joconte** [comment]
Todo file deleted

**Comment by rolfbjarne** [comment]
build

**Comment by dalexsoto** [comment]
build

**Comment by dalexsoto** [comment]
It just needed one last bit @joconte, we use framework.sources to track files in our build system, since this PR added `ITCompat.cs` we just need our build system to be aware of it.

**Comment by dalexsoto** [comment]
Thanks a lot for your contribution ❤️

**Pull request merged**